ArticleService.js

```diff
@@ -0,0 +1,188 @@
+class Article {
+
+    constructor(title, comment, writer, likeCount) {
+        this.title = title;
+        this.content = comment;
+        this.writer = writer;
+        this.likeCount = likeCount;
+        this.createdAt = new Date();
+    }
+
+    like() {
+        this.likeCount++;
+    }
+
+}
+
+export function getArticleList(page, pageSize, keyword, orderBy = 'recent') {
+
+    const url = new URL('https://panda-market-api-crud.vercel.app/articles');
+
+    if (page !== undefined && !isNaN(page)) {
+        url.searchParams.append('page', page);
+    }
+
+    if (pageSize !== undefined && !isNaN(pageSize)) {
+        url.searchParams.append('pageSize', pageSize);
+    }
+
+    if (keyword !== undefined) {
+        url.searchParams.append('keyword', `${keyword}`);
+    }
+
+    if (orderBy === 'recent' || orderBy === 'like') {
+        url.searchParams.append('orderBy', `${orderBy}`);
+    }
+
+    return fetch(url)
+        .then((response) => {
+            if (!response.ok) {
+                throw new Error('응답의 상태코드가 2XX가 아닙니다');
+            }
+            return response.json();
+        })
+        .then((data) => console.log(data))
+        .catch((error) => {
+            console.error(error)
+            throw error
+        });
+}
+
+export function getArticle(articleId = undefined) {
+
+    const url = new URL('https://panda-market-api-crud.vercel.app/articles');
+
+    if (articleId === undefined) { // ID가 없으면 얼리리턴, swagger에서 articleId가 required 값
+        console.error('articleID를 입력해주세요');
+        return;
+    }
+
+    if (isNaN(articleId)) {
+        console.error('articleId는 숫자를 입력해야합니다');
+        return;
+    }
+    url.pathname += `/${articleId}`; // url paht에 articleId 추가
+
+    return fetch(url)
+        .then((response) => {
+            if (!response.ok) {
+                throw new Error('응답의 상태코드가 2XX가 아닙니다');
+            }
+            return response.json();
+        })
+        .then((data) => console.log(data))
+        .catch((error) => {
+            console.error(error)
+            throw error
+        });
+
+}
+
+export function createArticle(title, content, image) {
+
+    const url = new URL('https://panda-market-api-crud.vercel.app/articles');
+
+    const createArticleObject = {
+        title: `${title}`,
+        content: `${content}`,
+        image: `${image}`
+    };
+
+    return fetch(url, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(createArticleObject),
+    })
+        .then((response) => {
+            if (!response.ok) {
+                throw new Error('응답의 상태코드가 2XX가 아닙니다');
+            }
+            return response.json();
+        })
+        .then((data) => console.log(data))
+        .catch((error) => {
+            console.error(error)
+            throw error
+        });

```

Comment:
이 코드가 대부분의 코드에서 반복되고있죠? 그러면 별도 함수로 추출해서 공통 로직을 하나의 함수로 분리할 수 있어요.

```suggestion
// 공통 fetch 핸들링 함수
async function fetchWithHandling(url, options = {}) {
    try {
        const response = await fetch(url, options);
        if (!response.ok) {
            if (response.status === 404) {
                throw new Error('404: 존재하지 않는 리소스입니다.');
            }
            throw new Error('응답의 상태코드가 2XX가 아닙니다');
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error(error);
        throw error;
    }
}

// 적용 예시: createArticle 함수
export async function createArticle(title, content, image) {
    const url = new URL('https://panda-market-api-crud.vercel.app/articles');
    const createArticleObject = {
        title: `${title}`,
        content: `${content}`,
        image: `${image}`
    };
    const options = {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(createArticleObject),
    };
    const data = await fetchWithHandling(url, options);
    console.log(data);
    return data;
}
```

Reply:
중복된 코드를 별도의 함수로 추출하면 코드의 재사용성과 유지보수성이 크게 향상됩니다! 반복되는 코드가 있다면 추출하도록 리팩토링해보면 좋아요